src/containers/blocks.jsx

```diff
@@ -97,9 +103,13 @@ class Blocks extends React.Component {
         if (prevProps.toolboxXML !== this.props.toolboxXML) {
             const categoryName = this.workspace.toolbox_.getSelectedCategoryName();
             const offset = this.workspace.toolbox_.getCategoryScrollOffset();
-            this.workspace.updateToolbox(this.props.toolboxXML);
-            const currentCategoryPos = this.workspace.toolbox_.getCategoryPositionByName(categoryName);
-            this.workspace.toolbox_.setFlyoutScrollPos(currentCategoryPos + offset);
+            // rather than update the toolbox "sync" -- update it in the next frame
+            clearTimeout(this.toolboxUpdate);
+            this.toolboxUpdate = setTimeout(() => {
+                this.workspace.updateToolbox(this.props.toolboxXML);
+                const currentCategoryPos = this.workspace.toolbox_.getCategoryPositionByName(categoryName);
+                this.workspace.toolbox_.setFlyoutScrollPos(currentCategoryPos + offset);
+            }, 0);
```

Comment:
Super minor: I'd consider renaming the variable so its purpose is more obvious. Maybe `toolboxUpdateId` or `pendingToolboxUpdateId` or `theGreatAmericanNovelAboutUpdatingToolboxesShallBeTrackedWithThisId` or something. That last one might be a bit verbose, though.

Reply:
`toolboxUpdateTimeout` is where I'd lean I think...